Panic with skaffold dev in 1.13.2

This is happening in 1.13.2 as well as latest build (b642a69 as of issue time). It does NOT happen in 1.13.1.

Given the panic, it appears to have been introduced by #4676.

### Expected behavior
* Logs start streaming and containers run normally

### Actual behavior
* Logs start streaming
* After a moment or two, skaffold panics
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17b3bea]

goroutine 394 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/trigger.(*pollTrigger).Start.func1(0xc000706230, 0xc000469be0, 0xc0006c4de0, 0x215bea0, 0xc0005ad380)
        /skaffold/pkg/skaffold/trigger/triggers.go:105 +0xda
created by github.com/GoogleContainerTools/skaffold/pkg/skaffold/trigger.(*pollTrigger).Start
        /skaffold/pkg/skaffold/trigger/triggers.go:99 +0x9f
```

### Information
- Skaffold version: 1.13.2 (also happens on b642a69)
- Local Cluster: k3d
- Operating system: Manjaro Linux
- Contents of skaffold.yaml: can't paste here, but at a high-level has some Jib builds, has some local Docker builds, deployments are using K8S manifests, and I'm launching using a profile

### Steps to reproduce the behavior

1. `skaffold dev -p <profile> --port-forward`

--- a/pkg/skaffold/trigger/triggers.go
+++ b/pkg/skaffold/trigger/triggers.go
@@ -75,6 +75,7 @@ func newFSNotifyTrigger(cfg Config, isActive func() bool) *fsNotifyTrigger {
 		Interval:   time.Duration(cfg.WatchPollInterval()) * time.Millisecond,
 		workspaces: workspaces,
 		isActive:   isActive,
+		watchFunc:  notify.Watch,
 	}
 }
 
@@ -179,6 +180,7 @@ type fsNotifyTrigger struct {
 	Interval   time.Duration
 	workspaces map[string]struct{}
 	isActive   func() bool
+	watchFunc  func(path string, c chan<- notify.EventInfo, events ...notify.Event) error
 }
 
 // Debounce tells the watcher to not debounce rapid sequence of changes.
@@ -206,7 +208,7 @@ func (t *fsNotifyTrigger) Start(ctx context.Context) (<-chan bool, error) {
 	}
 
 	// Watch current directory recursively
-	if err := notify.Watch(filepath.Join(wd, "..."), c, notify.All); err != nil {
+	if err := t.watchFunc(filepath.Join(wd, "..."), c, notify.All); err != nil {
 		return nil, err
 	}
 
@@ -216,7 +218,7 @@ func (t *fsNotifyTrigger) Start(ctx context.Context) (<-chan bool, error) {
 			continue
 		}
 
-		if err := notify.Watch(filepath.Join(wd, w, "..."), c, notify.All); err != nil {
+		if err := t.watchFunc(filepath.Join(wd, w, "..."), c, notify.All); err != nil {
 			return nil, err
 		}
 	}
@@ -267,6 +269,7 @@ func StartTrigger(ctx context.Context, t Trigger) (<-chan bool, error) {
 
 		t = &pollTrigger{
 			Interval: notifyTrigger.Interval,
+			isActive: notifyTrigger.isActive,
 		}
 		ret, err = t.Start(ctx)
 	}

--- a/pkg/skaffold/trigger/triggers_test.go
+++ b/pkg/skaffold/trigger/triggers_test.go
@@ -231,7 +231,7 @@ func TestStartTrigger(t *testing.T) {
 		{
 			description: "fsNotify trigger works",
 			trigger: &fsNotifyTrigger{
-				Interval:   2 * time.Second,
+				Interval:   200 * time.Millisecond,
 				workspaces: nil,
 				isActive:   func() bool { return false },
 				watchFunc: func(string, chan<- notify.EventInfo, ...notify.Event) error {

--- a/pkg/skaffold/trigger/triggers_test.go
+++ b/pkg/skaffold/trigger/triggers_test.go
@@ -18,10 +18,13 @@ package trigger
 
 import (
 	"bytes"
+	"context"
+	"fmt"
 	"testing"
 	"time"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/rjeczalik/notify"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/config"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/runcontext"
@@ -52,6 +55,7 @@ func TestNewTrigger(t *testing.T) {
 					"../workspace":            {},
 					"../some/other/workspace": {},
 				},
+				watchFunc: notify.Watch,
 			},
 		},
 		{
@@ -87,10 +91,20 @@ func TestNewTrigger(t *testing.T) {
 			got, err := NewTrigger(runCtx, nil)
 			t.CheckError(test.shouldErr, err)
 			if !test.shouldErr {
-				t.CheckDeepEqual(test.expected, got, cmp.AllowUnexported(fsNotifyTrigger{}), cmp.AllowUnexported(manualTrigger{}), cmp.AllowUnexported(pollTrigger{}))
+				t.CheckDeepEqual(test.expected, got, cmp.AllowUnexported(fsNotifyTrigger{}), cmp.Comparer(ignoreFuncComparer), cmp.AllowUnexported(manualTrigger{}), cmp.AllowUnexported(pollTrigger{}))
 			}
 		})
 	}
+}
+
+func ignoreFuncComparer(x, y func(path string, c chan<- notify.EventInfo, events ...notify.Event) error) bool {
+	if x == nil && y == nil {
+		return true
+	}
+	if x == nil || y == nil {
+		return false
+	}
+	return true // cannot assert function equality, so skip
 }
 
 func TestPollTrigger_Debounce(t *testing.T) {
@@ -206,5 +220,43 @@ func TestManualTrigger_LogWatchToUser(t *testing.T) {
 
 		got, want := out.String(), test.expected
 		testutil.CheckDeepEqual(t, want, got)
+	}
+}
+
+func TestStartTrigger(t *testing.T) {
+	tests := []struct {
+		description string
+		trigger     Trigger
+	}{
+		{
+			description: "fsNotify trigger works",
+			trigger: &fsNotifyTrigger{
+				Interval:   2 * time.Second,
+				workspaces: nil,
+				isActive:   func() bool { return false },
+				watchFunc: func(string, chan<- notify.EventInfo, ...notify.Event) error {
+					return nil
+				},
+			},
+		},
+		{
+			description: "fallback on polling trigger",
+			trigger: &fsNotifyTrigger{
+				Interval:   200 * time.Millisecond,
+				workspaces: nil,
+				isActive:   func() bool { return false },
+				watchFunc: func(string, chan<- notify.EventInfo, ...notify.Event) error {
+					return fmt.Errorf("failed to start watch trigger")
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		testutil.Run(t, test.description, func(t *testutil.T) {
+			_, err := StartTrigger(context.Background(), test.trigger)
+			time.Sleep(1 * time.Second)
+			t.CheckNoError(err)
+		})
 	}
 }

